libraries/AP_AS5600_AOA/AS5600_AOA.cpp

```diff
@@ -0,0 +1,266 @@
+/****************************************************
+ * AMS 5600 Angle of Attack Class for Ardupilot platform
+ * Author: Cole Mero
+ * Date: 15 Dec 2014
+ * File: AS5600_AOA.cpp
+ * Version 1.00
+ * www.ams.com
+ *
+ * Description:  This class has been designed to
+ * access the AS5600 magnetic encoder sensor to
+ * read the angle of attack and record it for experimental purposes
+ *
+ * It was adapted from the Arduino library available for the AS5600 sensor.
+ *
+***************************************************/
+
+#include <unistd.h>
+#include "AS5600_AOA.h"
+
+extern const AP_HAL::HAL &hal;
+
+/****************************************************
+ * Method: AS5600_AOA
+ * In: none
+ * Out: none
+ * Description: constructor for AMS 5600
+***************************************************/
+ AS5600_AOA::AS5600_AOA()
+{
+
+    bus = 1; //Sets the bus number for the device, unclear what this number should be, trial and error to make it work
+    address = 0x36; //This is the I2C address for the device, it is set by the manufacturer
+
+   /*load register values*/
+
+  _zmco = 0x00;

```

Comment:
```suggestion
  static const uint8_t REG_ZMCO = 0x00;
```

throughout here.

Also, these should be moved down to `private:` section in the header, or to the top of the `cpp` file.